src/components/Sidebar.jsx

```diff
@@ -0,0 +1,151 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import Drawer from "@mui/material/Drawer";
+import Toolbar from "@mui/material/Toolbar";
+import List from "@mui/material/List";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemIcon from "@mui/material/ListItemIcon";
+import ListItemText from "@mui/material/ListItemText";
+import ExpandLess from "@mui/icons-material/ExpandLess";
+import ExpandMore from "@mui/icons-material/ExpandMore";
+import Collapse from "@mui/material/Collapse";
+import { Box, Button, Divider } from "@mui/material";
+import LogoutIcon from "@mui/icons-material/Logout";
+import PieChartOutlinedIcon from "@mui/icons-material/PieChartOutlined";
+import GridViewOutlinedIcon from "@mui/icons-material/GridViewOutlined";
+import ReviewsRoundedIcon from "@mui/icons-material/ReviewsRounded";
+import SettingsOutlinedIcon from "@mui/icons-material/SettingsOutlined";
+
+const drawerWidth = 300;
+
+const tabList = [
+  {
+    name: "Overview",
+    path: "/overview",
+    icon: GridViewOutlinedIcon,
+  },
+  {
+    name: "History",
+    path: "/history",
+    icon: PieChartOutlinedIcon,
+  },
+  {
+    name: "Reviews",
+    path: "/reviews",
+    icon: ReviewsRoundedIcon,
+  },
+  {
+    name: "Restaurant",
+    path: "/restaurant",
+    icon: null,
+  },
+  {
+    name: "Profile",
+    path: "/hostProfile",
+    icon: null,
+  },
+];
+
+const Sidebar = (props) => {
+  const upperTabs = tabList.filter(
```

Comment:
Clean code: In order to write clean code, all the hooks should be on the top, then function
The performance would be better if you can separate the upper part and lower part right at the beginning instead of having two loops to separate it